sdk/eventhubs/azure-messaging-eventhubs/test/ut/processor_test.cpp

```diff
@@ -4,26 +4,457 @@
+      {
+        std::unique_lock<std::mutex> lock(m_stateLock);
+        m_waitables += count;
+      }
+      void CompleteWaiter()
+      {
+        std::unique_lock<std::mutex> lock(m_stateLock);
+        m_waitables -= 1;
+        if (m_waitables == 0)
+        {
+          m_waitComplete.notify_all();
+        }
+      }
+
+    private:
+      int32_t m_waitables{};
+      std::mutex m_stateLock;
+      std::condition_variable m_waitComplete;
+    };
   } // namespace
+
+  class ProcessorTest : public EventHubsTestBase {
+  protected:
+    static void SetUpTestSuite() { EventHubsTestBase::SetUpTestSuite(); }
+    static void TearDownTestSuite() { EventHubsTestBase::TearDownTestSuite(); }
+
+    void TearDown() override
+    {
+      EventHubsTestBase::TearDown();
+      // When the test is torn down, the global state MUST be idle. If it is not, something leaked.
+      Azure::Core::Amqp::Common::_detail::GlobalStateHolder::GlobalStateInstance()->AssertIdle();
+    }
+
+    void TestWithLoadBalancer(Models::ProcessorStrategy processorStrategy)
+    {
+      Azure::Core::Context context = Azure::Core::Context::ApplicationContext.WithDeadline(
+          Azure::DateTime::clock::now() + std::chrono::minutes(5));
+
+      std::string eventHubName{GetEnv("EVENTHUB_NAME")};
+      std::string consumerGroup = GetEnv("EVENTHUB_CONSUMER_GROUP");
+
+      std::string const connectionString = GetEnv("EVENTHUB_CONNECTION_STRING");
+      Azure::Messaging::EventHubs::ConsumerClientOptions consumerClientOptions;
+      consumerClientOptions.ApplicationID
+          = testing::UnitTest::GetInstance()->current_test_info()->name();
+      consumerClientOptions.Name = testing::UnitTest::GetInstance()->current_test_info()->name();
+
+      std::string containerName{GetRandomName("proctest")};
+
+      // Create the checkpoint store
+      std::shared_ptr<CheckpointStore> checkpointStore{std::make_shared<TestCheckpointStore>()};
+      GTEST_LOG_(INFO) << "Checkpoint store created";
+
+      std::shared_ptr<ConsumerClient> consumerClient{std::make_shared<ConsumerClient>(
+          connectionString, eventHubName, consumerGroup, consumerClientOptions)};
+      GTEST_LOG_(INFO) << "Consumer Client created";
+
+      ProcessorOptions processorOptions;
+      processorOptions.LoadBalancingStrategy = processorStrategy;
+      processorOptions.UpdateInterval = std::chrono::milliseconds(1000);
+      processorOptions.Prefetch = 1500; // Set the initial link credits to 1500.
+
+      Processor processor{consumerClient, checkpointStore, processorOptions};
+
+      // Warm up the consumer client - establish connection to the server, etc.
+      auto eventHubProperties = consumerClient->GetEventHubProperties(context);
+
+      ProducerClientOptions producerOptions;
+      producerOptions.Name = "Producer for LoadBalancerTest";
+      ProducerClient producerClient{connectionString, eventHubName, producerOptions};
+
+#if PROCESSOR_ON_TEST_THREAD
```

Comment:
Where does this gets defined?
Should it be prefixed with `_azure_AMQP_`?

Reply:
This gets enabled nowhere.

It 100% should NOT be `_AMQP_` because this is not AMQP - it's eventhubs.

And the _azure_ prefix should be restricted to private definitions in public headers. processor_test.cpp is not a public header.